### PR TITLE
feat(sfs): add a new resource to manage SFS Turbo DU task

### PR DESCRIPTION
**Pull request by ruwenqiang123**
**What this PR does / why we need it**:

Add a new resource to manage SFS Turbo DU task.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1.support a new resource to manage SFS Turbo DU task
2.support related document and acceptance test.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.
* [x] CheckDeleted

About the checking items of the CheckDeleted logic, we need you provide these corresponding verification results, and paste the screenshots of the results (if it is actually checked) to the designated location.

**a. During query operation (Read Context)**
  aa. Resource not found
  \>>>>>> Paste the screenshot here <<<<<<
image

  ab. Related resources (parent resources) not found
  \>>>>>> Paste the screenshot here <<<<<<
image
image
image

**b. During delete/disassociate/unbind operation (Delete Context)**
  ba. Resource not found
  \>>>>>> Paste the screenshot here <<<<<<
image
image

  bb. Related resources (parent resources) not found
  \>>>>>> Paste the screenshot here <<<<<<
image
image

## Acceptance Steps Performed

```
$ make testacc TEST="./huaweicloud/services/acceptance/sfs" TESTARGS="-run TestAccDuTask_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/sfs -v -run TestAccDuTask_basic -timeout 360m -parallel 4
=== RUN   TestAccDuTask_basic
=== PAUSE TestAccDuTask_basic
=== CONT  TestAccDuTask_basic
--- PASS: TestAccDuTask_basic (787.77s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/sfs       787.817s
```
```
$ make testacc TEST="./huaweicloud/services/acceptance/sfs" TESTARGS="-run TestAccDuTask_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/sfs -v -run TestAccDuTask_basic -timeout 360m -parallel 4
=== RUN   TestAccDuTask_basic
=== PAUSE TestAccDuTask_basic
=== CONT  TestAccDuTask_basic
--- PASS: TestAccDuTask_basic (774.66s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/sfs       774.736s
```

### CODE REVIEW

File: docs/resources/sfs_turbo_du_task.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+page_title: "HuaweiCloud: huaweicloud_sfs_turbo_du_task"
+description: ""
+---
+
+# huaweicloud_sfs_turbo_du_task
+
+Manages a DU task resource under the SFS Turbo within HuaweiCloud.
+
+-> SFS Turbo File systems created after August 1, 2023 support this resource.
+    If there are `10` executing tasks, no new tasks areallowed to be created.
```
Comment (deer-hang) [inline docs/resources/sfs_turbo_du_task.md:13]: Remove extra spaces.

File: docs/resources/sfs_turbo_du_task.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    It is recommended that no more than `4` query requests be submitted at the same time.
+    Querying a very large directory tasks a long time, please do not submit the request again.
```
Comment (deer-hang) [inline docs/resources/sfs_turbo_du_task.md]: Please do not use loose English and do not explain too many words that have nothing to do with resources.

File: docs/resources/sfs_turbo_du_task.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+-> The resource supported SFS Turbo file system types are **20MB/s/TiB**, **40MB/s/TiB**, **125MB/s/TiB**,
```
Comment (deer-hang) [inline docs/resources/sfs_turbo_du_task.md]: -> This resource is only available for the following SFS Turbo file system types: XXX

File: docs/resources/sfs_turbo_du_task.md (hunk 2 above)
Comment (deer-hang) [reply docs/resources/sfs_turbo_du_task.md]: Remove extra spaces.

File: docs/resources/sfs_turbo_du_task.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+* `status` - The status of the DU task. The valid values are as following:
+  + **SUCCESS**
+  + **DOING**
+  + **FAIL**
+
+* `dir_usage` - The direcrory resource usage (including subdirectories).
+  The [dir_usage](#du_task_dirUsage) structure is documented below.
+
+* `begin_time` - The start time of the DU task, in RFC3339 format.
```
Comment (deer-hang) [inline docs/resources/sfs_turbo_du_task.md:61]: What is the difference between UTC and RFC3339？

File: huaweicloud/services/sfs/resource_huaweicloud_sfs_turbo_du_task.go [hunk 5]
```diff
@@ -0,0 +1,294 @@
...
+		return diag.Errorf("error creating DU task: %s", err)
+	}
+
+	createDuTaskRespBody, err := utils.FlattenResponse(createDuTaskResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	taskId, err := jmespath.Search("task_id", createDuTaskRespBody)
+	if err != nil {
```
Comment (deer-hang) [inline huaweicloud/services/sfs/resource_huaweicloud_sfs_turbo_du_task.go]:
```suggestion
	if err != nil || taskId == nil {
```

File: huaweicloud/services/sfs/resource_huaweicloud_sfs_turbo_du_task.go [hunk 6]
```diff
@@ -0,0 +1,294 @@
...
+		}
+		return common.CheckDeletedDiag(d, err, "error retrieving DU task")
+	}
+
+	getDuTaskRespBody, err := utils.FlattenResponse(getDuTaskResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	beginTime := utils.ConvertTimeStrToNanoTimestamp(utils.PathSearch("begin_time", getDuTaskRespBody, nil).(string), "2006-01-02 15:04:05")
```
Comment (deer-hang) [inline huaweicloud/services/sfs/resource_huaweicloud_sfs_turbo_du_task.go]:
```suggestion
	beginTime := utils.ConvertTimeStrToNanoTimestamp(utils.PathSearch("begin_time", getDuTaskRespBody, "").(string), "2006-01-02 15:04:05")
```

File: huaweicloud/services/sfs/resource_huaweicloud_sfs_turbo_du_task.go [hunk 7]
```diff
@@ -0,0 +1,294 @@
...
+
+	return []map[string]interface{}{params}
+}
+
+func resourceDuTaskDelete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	client, err := cfg.SfsV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating SFS v1 Client: %s", err)
```
Comment (deer-hang) [inline huaweicloud/services/sfs/resource_huaweicloud_sfs_turbo_du_task.go]:
```suggestion
		return diag.Errorf("error creating SFS v1 client: %s", err)
```

File: docs/resources/sfs_turbo_du_task.md [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+
+# huaweicloud_sfs_turbo_du_task
+
+Manages a DU task resource under the SFS Turbo within HuaweiCloud.
+
+-> SFS Turbo File systems created after August 1, 2023 support this resource.
+  If there are `10` tasks being executed, then creating no new task is not allowed. If the number of concurrent
+  requests are oversize, then the file system performance may be affected. It is suggested that a maximum of `4`
+  query requests be sent at the same time.
+
```
Comment (deer-hang) [inline docs/resources/sfs_turbo_du_task.md:16]: -> Only SFS Turbo File systems created after August 1, 2023 can use this resource. Creating new task is not supported when there are 10 or more executing tasks under the system. In order not to affect the performance of the file system, it is recommended that the number of tasks submitted simultaneously under one system should not exceed 4.

File: docs/resources/sfs_turbo_du_task.md [hunk 9]
```diff
@@ -0,0 +1,99 @@
+---
+subcategory: "Scalable File Service (SFS)"
+layout: "huaweicloud"
+page_title: "HuaweiCloud: huaweicloud_sfs_turbo_du_task"
+description: ""
+---
```
Comment (jinyangyang222) [inline docs/resources/sfs_turbo_du_task.md:6]: please add `description`

File: huaweicloud/services/sfs/resource_huaweicloud_sfs_turbo_du_task.go [hunk 10]
```diff
@@ -0,0 +1,294 @@
...
+
+func resourceDuTaskImportState(_ context.Context, d *schema.ResourceData, _ interface{}) ([]*schema.ResourceData,
+	error) {
+	parts := strings.Split(d.Id(), "/")
+	if len(parts) != 2 {
+		return nil, fmt.Errorf("invalid format for import ID, want '<share_id>/<id>', but got '%s'", d.Id())
+	}
+
+	d.SetId(parts[1])
+	return []*schema.ResourceData{d}, d.Set("share_id", parts[0])
```
Comment (jinyangyang222) [inline huaweicloud/services/sfs/resource_huaweicloud_sfs_turbo_du_task.go:293]: Please handle any errors that may arise from setting the `share_id`

**Pull request merged**